gravitee-io/gravitee-access-management
# AM-835 : Key usage is always the one provided by the cert

Fix AM-835

## Description

JWKS exposed by the GW always display the keyUsage extracted by the Certificate even if the UI define another value.
UI settings is used only when the cert doesn't provide this info.

This PR change this behaviour to use in priority the settings provided by the UI if any.

## How to test

create cert.ext file with this content

```
[ v3_req ]

authorityKeyIdentifier=keyid,issuer
basicConstraints=CA:FALSE
keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
```
Then create a certificate.

```
openssl genrsa -out server.key 4096
openssl req -new -key server.key -out server.csr -sha256 -subj "/emailAddress=contact@graviteesource.com/CN=localhost/OU=Cockpit/O=GraviteeSource/L=Lille/ST=France/C=FR"
openssl x509 -req -in server.csr -set_serial 100 -extensions v3_req -days 6 -outform PEM -out server.cer -sha256 -passin pass:ca-secret -key server.key -extfile ./cert.ext
openssl pkcs12 -export -inkey server.key -in server.cer -out server.p12 -passout pass:server-secret -name am-server
```

Check that the cert contain keyUsage info:
```
$> openssl x509 -noout -ext keyUsage < server.cer 
X509v3 Key Usage: 
    Digital Signature, Non Repudiation, Key Encipherment, Data Encipherment
```

Then create a certificate at domain level using the server.p12 file 
* Keypass & storepass are the same : `server-secret`
* **DO NOT** check signature  or encryption option
![image](https://github.com/gravitee-io/gravitee-access-management/assets/3110552/13a77b89-93e2-4214-925b-ea534881349a)

Once created, you should see use value set to for the am-server object on this URL http://localhost:8092/domain/oidc/.well-known/jwks.json (to get the right jwks url go to the application endpoints tab)

![image](https://github.com/gravitee-io/gravitee-access-management/assets/3110552/fa5f87eb-7f4e-4dba-a0e8-a4f3ccb82176)

```
 "keys" : [ {
    "kty" : "RSA",
    "use" : "sig",
    "alg" : "RS512",
    "kid" : "am-server",
    "x5c" : [ ...
```

Then If you select encryption in the certificate configuration form, the use value should change to 'enc'
